### PR TITLE
search: graphql api to give detailed repo info

### PR DESCRIPTION
**Pull request by shimish2**
- DetailedRepoInfo graphql api returns detailed repo info given name.
- Each repo contains its manifests info.
- Each manifest entry contains digest, signature, tag and layers info
- Each layer info containes digest, size.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/search/common/oci_layout.go [hunk 1]
```diff
@@ -183,6 +221,72 @@ func (olu OciLayoutUtils) GetImageTagsWithTimestamp(repo string) ([]TagInfo, err
...
+	reposInfo := make([]RepoInfo, 0)
+
+	repos, err := olu.GetRepos()
+	if err != nil {
+		olu.Log.Error().Err(err).Msg("error getting all repositories list")
+
+		return reposInfo, err
+	}
+
+	for _, r := range repos {
```
Comment (rchincha) [inline pkg/extensions/search/common/oci_layout.go:234]: call it "repo"

**Review by rchincha — COMMENTED** [review]
Overall looks good, nitpicks only.

File: pkg/extensions/search/common/common_test.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 //go:build extended
 // +build extended
 
+// nolint:lll
```
[the remark this answers is not in the document]
Comment (shimish2) [reply pkg/extensions/search/common/common_test.go]: Removed nolint and fixed parts of code where linter was complaining.

File: pkg/extensions/search/common/oci_layout.go [hunk 3]
```diff
@@ -17,12 +18,30 @@ import (
 	"zotregistry.io/zot/pkg/storage"
 )
 
+const signedAnnotation = "dev.cosign.signature.baseimage"
```
[the remark this answers is not in the document]
Comment (shimish2) [reply pkg/extensions/search/common/oci_layout.go]: Done

File: pkg/extensions/search/common/oci_layout.go [hunk 4]
```diff
@@ -183,6 +202,65 @@ func (olu OciLayoutUtils) GetImageTagsWithTimestamp(repo string) ([]TagInfo, err
 	return tagsInfo, nil
 }
 
+func (olu OciLayoutUtils) GetRepoInfo(name string) (RepoInfo, error) {
```
[the remark this answers is not in the document]
Comment (shimish2) [reply pkg/extensions/search/common/oci_layout.go]: Using GetExpandedRepoInfo()

File: pkg/extensions/search/schema.graphql [hunk 5]
```diff
@@ -50,10 +50,27 @@ type ImageInfo {
...
+     Digest: String
+}
+
 type Query {
   CVEListForImage(image: String!) :CVEResultForImage 
   ImageListForCVE(id: String!) :[ImgResultForCVE]
   ImageListWithCVEFixed(id: String!, image: String!) :ImgResultForFixedCVE
   ImageListForDigest(id: String!) :[ImgResultForDigest]
   ImageListWithLatestTag:[ImageInfo]
+  DetailedRepoInfo(repo: String!):RepoInfo
```
[the remark this answers is not in the document]
Comment (shimish2) [reply pkg/extensions/search/schema.graphql]: Using ExpandedRepoInfo

File: pkg/extensions/search/common/common_test.go [hunk 6]
```diff
@@ -36,6 +36,15 @@ type ImgResponsWithLatestTag struct {
 	Errors               []ErrorGQL           `json:"errors"`
 }
 
+type DetailedRepoInfoResp struct {
```
Comment (rchincha) [inline pkg/extensions/search/common/common_test.go:39]: Do you want to keep these as "Detailed"? Maybe change these too?

Comment (rchincha) [reply pkg/extensions/search/common/common_test.go:39]: Want to use one keyword, else confusing to code readers

Comment (shimish2) [reply pkg/extensions/search/common/common_test.go]: Done

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**